mm2src/coins/utxo/utxo_common.rs

```diff
@@ -264,17 +267,21 @@ where
 /// Note `gas_fee` should be enough to execute all of the contract calls within UTXO outputs.
 /// QRC20 specific: `gas_fee` should be calculated by: gas_limit * gas_price * (count of contract calls),
 /// or should be sum of gas fee of all contract calls.
+///
+/// Please note `force_change_output` may cause an error if the change is less than dust.
 pub async fn generate_transaction<T>(
     coin: &T,
     utxos: Vec<UnspentInfo>,
     outputs: Vec<TransactionOutput>,
     fee_policy: FeePolicy,
     fee: Option<ActualTxFee>,
     gas_fee: Option<u64>,
+    force_change_output: bool,
```

Comment:
> If not, we can increase the fee accordingly outside of this fn

It may cause an incorrect calculation, because if the `sum_inputs - sum_outputs - fee(tx_size) < fee(change_output)`, then we should add another unspent input. It is a duplication of the logic of the `generate_transaction` fn that I wanted to avoid :D

I guess, we can just add `fee(change_output)` and (if the change is less than the `fee(change_output)`) add the `fee(another input)`

Reply:
> sum_inputs - sum_outputs - fee(tx_size) < fee(change_output), then we should add another unspent input.

Yeah, tricky :slightly_smiling_face: I guess this argument is required only when the coin has a dynamic fee. What if we simply increase the sat per byte rate quite significantly upon estimation? It should cover the possibility of change output to appear on a swap. Related to https://github.com/KomodoPlatform/atomicDEX-API/pull/797#discussion_r565869391

Reply:
E.g. by 10% instead of 1%.

Reply:
As I said in a comment above
> I found out the dynamic fee varies within 0.1 percent per block. I think if we set this value to 10%, it will be too much